### PR TITLE
Add needed imports in preprocess....

### PR DESCRIPTION
**Pull request by user1234554321**
- Created a parallel preprocess file
- Removed hardcoded '3' in Stratified K-folds
- sample_submission.csv to be passed as argument

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -220,14 +228,15 @@
...
+    model = nn.DataParallel(model)
     
     # create optimizer depending on the experiment running
     if args.chipped:
-        group1 = model.resnet.parameters()
-        group2 = list(model.attn_fc1.parameters()) + \
-                 list(model.attn_fc2.parameters()) + \
-                 list(model.attn_bn1.parameters())
-        group3 = list(model.fc4.parameters()) + list(model.bn4.parameters())
+        group1 = model.module.resnet.parameters()
```
Comment (CCareaga) [inline train.py:235]: was the model code changed to reflect this? what is the "module" field?

File: train.py [hunk 2]
```diff
@@ -220,14 +228,15 @@
     resnet = resnet_dict[args.resnet](pretrained=True)
     model = model_class(resnet, **model_kwargs)
     model = model.cuda()
+    model = nn.DataParallel(model)
```
Comment (CCareaga) [inline train.py:231]: What is the DataParallel wrapper for?

**Review by CCareaga — COMMENTED** [review]
I added a few comments addressing some changes, I want to minimize large  changes to keep the repo general so that others can easily use it.

File: train.py (hunk 2 above)
Comment (user1234554321) [reply train.py:231]: Hey, sorry, I forgot to mention....I used DataParallel to train on multiple gpus...should work with a single gpu as well as it wraps over the model....

Comment (user1234554321) [reply train.py:231]: I can remove it

Comment (CCareaga) [reply train.py:231]: That's fine, what about the alterations to the model code?

Comment (user1234554321) [reply train.py:231]: Pretty sure, no alterations required as it wraps it. I suggest to have a look at a wonderful tutorial by pytorch on nn.DataParallel; [Data Parallel Tutorial Pytorch](https://pytorch.org/tutorials/beginner/blitz/data_parallel_tutorial.html)

Comment (CCareaga) [reply train.py:231]: Ohhh! I see the `.module` is because of the DataParallel wrapper, gotcha.

**Pull request merged**